rskj-core/src/main/java/org/ethereum/db/MutableRepository.java

```diff
@@ -25,19 +25,29 @@
 import co.rsk.crypto.Keccak256;
 import co.rsk.db.MutableTrieCache;
 import co.rsk.db.MutableTrieImpl;
-import co.rsk.trie.*;
+import co.rsk.trie.IterationElement;
+import co.rsk.trie.MutableTrie;
+import co.rsk.trie.Trie;
+import co.rsk.trie.TrieKeySlice;
+import co.rsk.trie.TrieStore;
+import co.rsk.trie.TrieStoreImpl;
 import com.google.common.annotations.VisibleForTesting;
 import org.ethereum.core.AccountState;
 import org.ethereum.core.Repository;
 import org.ethereum.crypto.HashUtil;
 import org.ethereum.crypto.Keccak256Helper;
+import org.ethereum.datasource.HashMapDB;
 import org.ethereum.vm.DataWord;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
 import java.math.BigInteger;
-import java.util.*;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.Optional;
+import java.util.Set;
 
 public class MutableRepository implements Repository {
```

Comment:
I don't fully understand why the transient storage is implemented into the standard storage.

Reply:
This is not the standard storage, it's the mutable storage that if a transaction is not rolled back it can be reverted. The transient storage it's also mutable and the lifecycle of it it's the same for the regular mutable storage. That's why it's implemented here, for the sake of simplicity and to be managed in the same way. It will get advantage of the whole lifecycle of the mutable storage that we already have.

Reply:
ok, it is interesting point of view.